File: python/stacks/StaticStack.py
class StaticStack:
    def __init__(self) -> None:
        self.__pointer = -1
        self.__DataArray = []
 
    def isEmpty(self) -> bool:
        return self.__pointer == -1
  
    def showData(self):
        if self.isEmpty():
            print("Stack is empty!")
        else:
            print("All stack data: ")
            for i, item in enumerate(self.__DataArray):
                if i > self.__pointer:
                    break
                print(item)

    def Append(self, obj):
        self.__pointer += 1
        if 0 <= self.__pointer < len(self.__DataArray):
            self.__DataArray[self.__pointer] = obj
        self.__DataArray.append(obj)

    def Pop(self):
        if not self.isEmpty():
            self.__pointer -= 1

    def Top(self):
        if not self.isEmpty():
            print(self.__DataArray[self.__pointer])


 

File: Java/Stacks/StatickStack/StaticStack.java
package Java.Stacks.StatickStack;

public class StaticStack implements StackInterface{
  private Object[] data;
  private int head;

  StaticStack() {
    this(10);
  }

  StaticStack(int x) {
    this.data = new Object[x];
    this.head = -1;
  }

  public Boolean isEmpty() {
    return this.head == -1;
  }

  public Boolean isFull() {
    return this.head == this.data.length - 1;
  }

  public String Top() {
    if (!isEmpty()) {
      return data[head].toString();
    }
    return null;
  }

  public void Append(Object obj) {
    if (!isFull()) {
      head++;
      data[head] = obj;
    }
  }

  public void Pop() {
    if(!isEmpty()) {
      head--;
    }
  }
}

